templates/master/00-master/gcp/files/etc-kubernetes-manifests-gcp-routes-controller.yaml

```diff
@@ -0,0 +1,35 @@
+filesystem: "root"
+mode: 0644
+path: "/etc/kubernetes/manifests/gcp-routes-controller.yaml"
+contents:
+  inline: |
+    apiVersion: v1
+    kind: Pod
+    metadata:
+      name: gcp-routes-controller
+      namespace: kube-system
+    spec:
+      containers:
+      - name: gcp-routes-controller
+        image: "{{.Images.gcpRoutesControllerKey}}"
+        command: ["gcp-routes-controller"]
+        args:
+        - "run"
+        - "--health-check-url=https://127.0.0.1:6443/healthz"
```

Comment:
shouldn't this be /readyz? We want an early signal in order to not drop traffic or route it wrongly.

Reply:
@abhinavdahiya ptal

Reply:
> shouldn't this be /readyz? We want an early signal in order to not drop traffic or route it wrongly.

https://github.com/openshift/installer/blob/e8b75d0bd2a5c254fa3be8d1fadee902301ca4e3/data/data/gcp/network/lb-private.tf#L12
https://github.com/openshift/installer/blob/e8b75d0bd2a5c254fa3be8d1fadee902301ca4e3/data/data/aws/vpc/master-elb.tf#L66

We use the readyz for the LB... for graceful shutdown. the routes-controller is meant to open the egress to other apiservers when we know the one on this is no longer running, not just accepting new connections..

but i'm fine with what the apiserver team recommends.

Reply:
I don't see a difference to the LB. With readyz we report "don't send me traffic". This is the same for local clients. With healthz you will always have a race because when it fails, it is too late already, possibly for seconds.

Reply:
created https://github.com/openshift/machine-config-operator/pull/1417 - please let me know about BZs and I'll go ahead and create them